### PR TITLE
Add optimized sensor management tutorial

### PR DESCRIPTION
**Pull request by nperree-dstl**
Add tutorial explaining the sensor managers which use the Scipy optimize library. This PR also includes some updates/improvements to the optimised sensor managers.

### CODE REVIEW

File: docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+# ^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Each sensor management method requires a timestamp and a list of tracks at each time step when calling
+# the function :meth:`choose_actions`. This returns a mapping of sensors and actions to be taken by each
+# sensor, decided by the sensor managers.
+#
+# For each sensor management method, at each time step the chosen action is given to the sensors and then
+# measurements taken. The tracks are updated based on these measurements with predictions made for tracks
+# which have not been observed.
+#
```
Comment (hpritchett-dstl) [inline docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py:296]: The paragraph above loosely implies that predictions haven't been made for the tracks that are observed. A minor re-wording could help clarify this.

File: docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py [hunk 2]
```diff
@@ -0,0 +1,631 @@
...
+axes[1].plot(times, [metric.value for metric in va_metricA.value],
+             label='BruteForceSensorManager')
+axes[1].plot(times, [metric.value for metric in va_metricB.value],
+             label='OptimizeBruteSensorManager')
+axes[1].plot(times, [metric.value for metric in va_metricC.value],
+             label='OptimizeBasinHoppingSensorManager')
+axes[1].legend()
+
+# %%
+# Uncertainty metric
```
Comment (hpritchett-dstl) [inline docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py:580]: I feel like this plot could be made more impactful by reducing the covariance in the prior states, or by making the sensor models more noisy. This is something that's subjective though, so feel free to ignore me!

Comment (nperree-dstl) [reply docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py:580]: Is this for the uncertainty metric graph? Good point, I may also remove the OSPA one as it doesn't show anything useful

Comment (hpritchett-dstl) [reply docs/tutorials/sensormanagement/03_OptimisedSensorManagement.py:580]: Yes, the uncertainty metric plot.

**Review by hpritchett-dstl — APPROVED** [review]

File: stonesoup/sensormanager/optimise.py [hunk 3]
```diff
@@ -37,67 +37,120 @@ def optimise_func(x):
...
         result = brute(optimise_func,
                        ranges=ranges,
-                       Ns=self.number_of_grid_points)
+                       Ns=self.n_grid_points,
+                       full_output=self.full_output,
+                       finish=self.finish_func,
+                       disp=self.disp)
+
+        if self.full_output:
+            print('Full output:', result)
```
Comment (sdhiscocks) [inline stonesoup/sensormanager/optimise.py:102]:
```suggestion
```

File: stonesoup/sensormanager/optimise.py [hunk 4]
```diff
@@ -37,67 +37,120 @@ def optimise_func(x):
...
+                            'accepted.')
+    stepsize: float = Property(default=0.5,
+                               doc='Maximum step size for use in the random displacement.')
+    interval: int = Property(default=50,
+                             doc='Interval for how often to update the stepsize.')
+    disp: bool = Property(default=False,
+                          doc='Set to True to print status messages.')
+    niter_success: int = Property(default=None,
+                                  doc='Stop the run if the global minimum candidate '
+                                      'remains the same'
```
Comment (sdhiscocks) [inline stonesoup/sensormanager/optimise.py:140]:
```suggestion
                                      'remains the same '
```

**Review by sdhiscocks — CHANGES_REQUESTED** [review]

**Review by sdhiscocks — APPROVED** [review]

**Pull request merged**